sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -153,6 +154,12 @@ object Literal {
     Literal(CatalystTypeConverters.convertToCatalyst(v), dataType)
   }
 
+  def create[T : TypeTag](v: T): Literal = {
+    val ScalaReflection.Schema(dataType, _) = ScalaReflection.schemaFor[T]
+    val convert = CatalystTypeConverters.createToCatalystConverter(dataType)
```

Comment:
Oh, my bad. I found the code you suggested made some tests fail. Actually, `CatalystTypeConverters.convertToCatalyst` does not do the same thing with `CatalystTypeConverters.createToCatalystConverter(dataType)`. For instance, `CatalystTypeConverters.convertToCatalyst` does not convert `TupleX` to `GenericInternalRow`.